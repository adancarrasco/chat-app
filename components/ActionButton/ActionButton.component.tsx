import React, { useState } from "react";
import { StyleSheet, Text, Animated } from "react-native";

type ActionButtonProps = {
  isBottomButton?: boolean;
  onTouchEnd: () => void;
  children?: HTMLElement | string;
};

export default function ActionButton({
  isBottomButton,
  onTouchEnd,
  children
}: ActionButtonProps) {
  const [fadeAnim, setFadeAnim] = useState(new Animated.Value(0));
  React.useEffect(() => {
    Animated.timing(fadeAnim, {
      toValue: 1,
      duration: 700,
      delay: 2500
    }).start();
  }, [fadeAnim]);
  return (
    <Animated.View
      style={[
        styles.container,
        isBottomButton ? { top: "50%" } : { top: 0 },
        { opacity: fadeAnim }
      ]}
      onTouchEnd={() => {
        onTouchEnd();
        setFadeAnim(new Animated.Value(0));
      }}
    >
      <Text>{children}</Text>
    </Animated.View>
  );
}

const styles = StyleSheet.create({
  container: {
    flexDirection: "row",
    padding: 10,
    borderRadius: 8,
    alignItems: "flex-end",
    justifyContent: "center",
    position: "absolute",
    zIndex: 10,
    width: "100%",
    height: "50%",
    opacity: 0
  }
});
