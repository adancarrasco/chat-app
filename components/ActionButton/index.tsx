export { default } from "./ActionButton.component";
