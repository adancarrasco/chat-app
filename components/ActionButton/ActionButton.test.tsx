import React from "react";
import renderer from "react-test-renderer";
import ActionButton from "./ActionButton.component";

describe("<ActionButton /> rendering", () => {
  const mockOnTouchEnd = () => {};
  it("Matches snapshot", () => {
    const tree = renderer
      .create(<ActionButton onTouchEnd={mockOnTouchEnd} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("Matches snapshot with children", () => {
    const tree = renderer
      .create(<ActionButton onTouchEnd={mockOnTouchEnd}>Hello</ActionButton>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
