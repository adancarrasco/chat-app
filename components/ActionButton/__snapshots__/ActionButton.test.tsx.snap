// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActionButton /> rendering Matches snapshot 1`] = `
<View
  onTouchEnd={[Function]}
  style={
    Object {
      "alignItems": "flex-end",
      "borderRadius": 8,
      "flexDirection": "row",
      "height": "50%",
      "justifyContent": "center",
      "opacity": 0,
      "padding": 10,
      "position": "absolute",
      "top": 0,
      "width": "100%",
      "zIndex": 10,
    }
  }
>
  <Text />
</View>
`;

exports[`<ActionButton /> rendering Matches snapshot with children 1`] = `
<View
  onTouchEnd={[Function]}
  style={
    Object {
      "alignItems": "flex-end",
      "borderRadius": 8,
      "flexDirection": "row",
      "height": "50%",
      "justifyContent": "center",
      "opacity": 0,
      "padding": 10,
      "position": "absolute",
      "top": 0,
      "width": "100%",
      "zIndex": 10,
    }
  }
>
  <Text>
    Hello
  </Text>
</View>
`;
